internals/daemon/daemon.go

```diff
@@ -720,8 +726,13 @@ func SetRebootMode(mode RebootMode) {
 	switch mode {
 	case SystemdMode:
 		rebootHandler = systemdModeReboot
+		fallbackRebootHandler = systemdModeReboot
```

Comment:
@anpep I think this proposal is solid. Just one other option which may result in slightly less code changes:

Instead of added ```fallbackRebootHandler```, you could record the ```rebootMode```. In this model, you only ever have one ```rebootHandler```, and then for both ```SyscallMode``` and ```ExternalMode``` the handler is set to the syscall handler. 

Then the only other thing you do is in ```doReboot```, just add an early return with your special error code:

```go
func (d *Daemon) doReboot(sigCh chan<- os.Signal, waitTimeout time.Duration) error {
    if rebootMode == ExternalMode {
        return ErrRestartExternal
    }
:
}
```